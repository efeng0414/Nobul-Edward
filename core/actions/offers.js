import {
  OFFERS_IS_LOADING,
  GET_JOB_OFFERS_SUCCESS,
  GET_JOB_OFFERS_FAILURE,
  GET_CONSUMER_BUY_OFFERS_SUCCESS,
  GET_CONSUMER_SELL_OFFERS_SUCCESS,
  GET_CONSUMER_BUY_OFFERS_FAILURE,
  GET_CONSUMER_SELL_OFFERS_FAILURE,
  GET_AGENT_BUY_OFFERS_SUCCESS,
  GET_AGENT_BUY_OFFERS_FAILURE,
  GET_AGENT_SELL_OFFERS_SUCCESS,
  GET_AGENT_SELL_OFFERS_FAILURE,
  GET_AGENT_AUTOBID_OFFERS_SUCCESS,
  GET_AGENT_AUTOBID_OFFERS_FAILURE,
  GET_AGENT_TO_CONTRACT_SELL_OFFERS_SUCCESS,
  GET_AGENT_TO_CONTRACT_BUY_OFFERS_SUCCESS,
  GET_OFFER_DETAIL_SUCCESS,
  GET_OFFER_DETAIL_FAILURE,
  GET_JOB_WINNING_OFFER_SUCCESS,
  GET_JOB_WINNING_OFFER_FAILURE,
  SAVE_OFFER_SUCCESS,
  SAVE_OFFER_FAILURE,
  ACCEPT_OFFER_SUCCESS,
  ACCEPT_OFFER_FAILURE,
  WITHDRAW_OFFER_SUCCESS,
  WITHDRAW_OFFER_FAILURE,
  PROMOTE_OFFER_SUCCESS,
  PROMOTE_OFFER_FAILURE,
  REJECT_OFFER_SUCCESS,
  REJECT_OFFER_FAILURE,
  SET_OFFER_HAS_BEEN_READ_SUCCESS,
  SET_OFFER_HAS_BEEN_READ_FAILURE,
  SET_FAVORITE_OFFER_SUCCESS,
  SET_FAVORITE_OFFER_FAILURE,
  CLEAR_WINNING_OFFER,
  SET_OFFERS_FILTERS,
  FETCH_OFFER_SUCCESS,
  FETCH_OFFER_FAILURE,
  CLEAR_BUY_OFFERS,
  CLEAR_SELL_OFFERS,
  GET_OFFERS_CREATED_AT_RANGE_SUCCESS,
  GET_OFFERS_CREATED_AT_RANGE_FAILURE,

  //__________________________________________
  // TODO:
  // check if these are being used
  GET_WINNING_OFFER_SUCCESS,
  GET_WINNING_OFFER_FAILURE,
  EDIT_BUY_OFFER_SUCCESS,
  EDIT_BUY_OFFER_FAILURE,
  EDIT_SELL_OFFER_SUCCESS,
  EDIT_SELL_OFFER_FAILURE,
  UPDATE_BUY_OFFER_SUCCESS,
  UPDATE_BUY_OFFER_FAILURE,
  UPDATE_SELL_OFFER_SUCCESS,
  UPDATE_SELL_OFFER_FAILURE
} from "../types/offers";

const offersIsLoading = ({ isLoading }) => ({
  type: OFFERS_IS_LOADING,
  payload: isLoading
});

const getJobOffersSuccess = ({ jobType, jobOffers }) => ({
  type: GET_JOB_OFFERS_SUCCESS,
  payload: {
    jobType,
    jobOffers
  }
});

const getJobOffersFailure = ({ error }) => ({
  type: GET_JOB_OFFERS_FAILURE,
  payload: error
});

const getConsumerBuyOffersSuccess = ({ payload }) => ({
  type: GET_CONSUMER_BUY_OFFERS_SUCCESS,
  payload
});

const getConsumerSellOffersSuccess = ({ payload }) => ({
  type: GET_CONSUMER_SELL_OFFERS_SUCCESS,
  payload
});

const getConsumerBuyOffersFailure = ({ error }) => ({
  type: GET_CONSUMER_BUY_OFFERS_FAILURE,
  payload: error
});

const getConsumerSellOffersFailure = ({ error }) => ({
  type: GET_CONSUMER_SELL_OFFERS_FAILURE,
  payload: error
});

const getAgentBuyOffersSuccess = ({ buyOffers }) => ({
  type: GET_AGENT_BUY_OFFERS_SUCCESS,
  payload: buyOffers
});

const getAgentBuyToContractOffersSuccess = ({ offers }) => ({
  type: GET_AGENT_TO_CONTRACT_BUY_OFFERS_SUCCESS,
  payload: offers
});

const getAgentSellToContractOffersSuccess = ({ offers }) => ({
  type: GET_AGENT_TO_CONTRACT_SELL_OFFERS_SUCCESS,
  payload: offers
});

const getAgentBuyOffersFailure = ({ error }) => ({
  type: GET_AGENT_BUY_OFFERS_FAILURE,
  payload: error
});

const getAgentSellOffersSuccess = ({ sellOffers }) => ({
  type: GET_AGENT_SELL_OFFERS_SUCCESS,
  payload: sellOffers
});

const getAgentSellOffersFailure = ({ error }) => ({
  type: GET_AGENT_SELL_OFFERS_FAILURE,
  payload: error
});

const getAgentAutobidOffersSuccess = ({ autobidOffers }) => ({
  type: GET_AGENT_AUTOBID_OFFERS_SUCCESS,
  payload: autobidOffers
});

const getAgentAutobidOffersFailure = ({ error }) => ({
  type: GET_AGENT_AUTOBID_OFFERS_FAILURE,
  payload: error
});

const getOfferDetailSuccess = ({ offerDetail }) => ({
  type: GET_OFFER_DETAIL_SUCCESS,
  payload: offerDetail
});

const getOfferDetailFailure = ({ error }) => ({
  type: GET_OFFER_DETAIL_FAILURE,
  payload: error
});

// TODO: save the offer on the store
const saveOfferSuccess = ({ offerId, jobType, offer }) => {
  return {
    type: SAVE_OFFER_SUCCESS,
    meta: {
      offerId,
      jobType
    },
    payload: { ...offer, jobType, offerId }
  };
};

const saveOfferFailure = ({ error }) => ({
  type: SAVE_OFFER_FAILURE,
  payload: error
});

const acceptOfferSuccess = () => ({
  type: ACCEPT_OFFER_SUCCESS
});

const acceptOfferFailure = ({ error }) => ({
  type: ACCEPT_OFFER_FAILURE,
  payload: { error }
});

const rejectOfferSuccess = () => ({
  type: REJECT_OFFER_SUCCESS
});

const rejectOfferFailure = ({ error }) => ({
  type: REJECT_OFFER_FAILURE,
  payload: { error }
});

const withdrawOfferSuccess = () => ({
  type: WITHDRAW_OFFER_SUCCESS
});

const withdrawOfferFailure = error => ({
  type: WITHDRAW_OFFER_FAILURE,
  payload: { error }
});

const promoteOfferSuccess = () => ({
  type: PROMOTE_OFFER_SUCCESS
});

const promoteOfferFailure = error => ({
  type: PROMOTE_OFFER_FAILURE,
  payload: { error }
});

const setOfferHasBeenReadSuccess = () => ({
  type: SET_OFFER_HAS_BEEN_READ_SUCCESS
});

const setOfferHasBeenReadFailure = ({ error }) => ({
  type: SET_OFFER_HAS_BEEN_READ_FAILURE,
  payload: { error }
});

const setFavoriteOfferSuccess = () => ({
  type: SET_FAVORITE_OFFER_SUCCESS
});

const setFavoriteOfferFailure = ({ error }) => ({
  type: SET_FAVORITE_OFFER_FAILURE,
  payload: { error }
});

const setOffersFilters = ({ filters }) => ({
  type: SET_OFFERS_FILTERS,
  payload: { filters }
});

const deleteWinningOffer = () => ({
  type: CLEAR_WINNING_OFFER,
  payload: { winningOffer: {} }
});

const getJobWinningOfferSuccess = ({ offer }) => ({
  type: GET_JOB_WINNING_OFFER_SUCCESS,
  payload: { offer }
});

const getJobWinningOfferFailure = ({ error }) => ({
  type: GET_JOB_WINNING_OFFER_FAILURE,
  payload: { error }
});

//__________________________________________

const getWinningOfferSuccess = ({ offer }) => ({
  type: GET_WINNING_OFFER_SUCCESS,
  payload: { winningOffer: offer }
});

const getWinningOfferFailure = ({ error }) => ({
  type: GET_WINNING_OFFER_FAILURE,
  payload: error
});

const editBuyOfferSuccess = buyOffer => {
  return { type: EDIT_BUY_OFFER_SUCCESS, payload: buyOffer };
};

const editBuyOfferFailure = error => {
  return {
    type: EDIT_BUY_OFFER_FAILURE,
    payload: error
  };
};

const editSellOfferSuccess = sellOffer => {
  return { type: EDIT_SELL_OFFER_SUCCESS, payload: sellOffer };
};

const editSellOfferFailure = error => {
  return {
    type: EDIT_SELL_OFFER_FAILURE,
    payload: error
  };
};

const updateBuyOfferSuccess = ({ payload }) => {
  return { type: UPDATE_BUY_OFFER_SUCCESS, payload };
};

const updateBuyOfferFailure = error => {
  return {
    type: UPDATE_BUY_OFFER_FAILURE,
    payload: error
  };
};

const updateSellOfferSuccess = ({ payload }) => {
  return { type: UPDATE_SELL_OFFER_SUCCESS, payload };
};

const updateSellOfferFailure = error => {
  return {
    type: UPDATE_SELL_OFFER_FAILURE,
    payload: error
  };
};

const fetchOfferSuccess = offer => {
  return {
    type: FETCH_OFFER_SUCCESS,
    payload: offer
  };
};

const fetchOfferFailure = error => {
  return {
    type: FETCH_OFFER_FAILURE,
    payload: error
  };
};

const clearBuyOffers = () => {
  return {
    type: CLEAR_SELL_OFFERS
  };
};

const clearSellOffers = () => {
  return {
    type: CLEAR_BUY_OFFERS
  };
};

const getOffersCreatedAtRangeSuccess = ({ offers }) => ({
  type: GET_OFFERS_CREATED_AT_RANGE_SUCCESS,
  payload: offers
});
const getOffersCreatedAtRangeFailure = ({ error }) => ({
  type: GET_OFFERS_CREATED_AT_RANGE_FAILURE,
  payload: error
});

export {
  offersIsLoading,
  getJobOffersSuccess,
  getJobOffersFailure,
  getConsumerBuyOffersSuccess,
  getConsumerBuyOffersFailure,
  getConsumerSellOffersSuccess,
  getConsumerSellOffersFailure,
  getAgentBuyOffersSuccess,
  getAgentBuyOffersFailure,
  getAgentSellOffersSuccess,
  getAgentSellOffersFailure,
  getAgentAutobidOffersSuccess,
  getAgentAutobidOffersFailure,
  getOfferDetailSuccess,
  getOfferDetailFailure,
  saveOfferSuccess,
  saveOfferFailure,
  acceptOfferSuccess,
  acceptOfferFailure,
  rejectOfferSuccess,
  rejectOfferFailure,
  withdrawOfferSuccess,
  withdrawOfferFailure,
  promoteOfferSuccess,
  promoteOfferFailure,
  setOfferHasBeenReadSuccess,
  setOfferHasBeenReadFailure,
  setFavoriteOfferSuccess,
  setFavoriteOfferFailure,
  setOffersFilters,
  deleteWinningOffer,
  getJobWinningOfferSuccess,
  getJobWinningOfferFailure,
  getAgentBuyToContractOffersSuccess,
  getAgentSellToContractOffersSuccess,
  clearSellOffers,
  clearBuyOffers,
  getOffersCreatedAtRangeSuccess,
  getOffersCreatedAtRangeFailure,
  //__________________________________________

  getWinningOfferSuccess,
  getWinningOfferFailure,
  editBuyOfferSuccess,
  editBuyOfferFailure,
  editSellOfferSuccess,
  editSellOfferFailure,
  updateBuyOfferSuccess,
  updateBuyOfferFailure,
  updateSellOfferSuccess,
  updateSellOfferFailure,
  fetchOfferSuccess,
  fetchOfferFailure
};
