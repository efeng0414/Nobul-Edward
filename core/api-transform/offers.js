export const CREATED_AT = "createdAt";
export const UPDATED_AT = "updatedAt";
export const CREATED_ON = "createdOn";
export const BUY = "buy";
export const SELL = "sell";
export const STATUS = "status";
export const AGENT_ID = "agentId";
export const COOPERATING_COMMISSION = "cooperatingCommission";
export const LISTING_COMMISSION = "listingCommission";
export const REBATE_COMMISSION = "rebateCommission";
export const CONSUMER_ID = "consumerId";
export const JOB_ID = "jobId";
export const PERSONALIZED_MESSAGE = "personalizedMessage";
export const TAGLINE = "tagline";
export const PROMOTION = "promotion";
export const PROMOTION_CHARGE_ID = "promotionChargeId";
export const SERVICES = "services";
export const CONSUMER_HAS_READ = "consumerHasRead";
export const IS_FAVORITE = "isFavorite";
export const IS_PROMOTED = "isPromoted";
export const IS_AUTOBID = "isAutobid";
export const PROMOTED_AT = "promotedAt";
