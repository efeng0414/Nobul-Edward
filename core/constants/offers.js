// offer status
export const OFFER_PENDING_VERIFICATION = "pendingVerification";
export const OFFER_PENDING_VERIFICATION_REJECTED =
  "pendingVerificationRejected";
export const OFFER_OPEN = "open";
export const OFFER_ACCEPTED = "accepted";
export const OFFER_ACCEPTED_CONFIRMED = "confirmedAccepted";
export const OFFER_NOT_ACCEPTED = "notAccepted";
export const OFFER_REJECTED = "rejected";
export const OFFER_JOBDELETED = "jobDeleted";
export const OFFER_JOBEXPIRED = "jobExpired";
export const OFFER_WITHDRAWN = "withdrawn";
// .offer status

export const OFFER_UNREAD = "unread";
export const OFFER_FAVORITE = "favorite";
export const ACCEPT_OFFER = "accept";
export const REJECT_OFFER = "reject";
export const OFFER_IS_PROMOTED = "isPromoted";

export const OFFER_LANDING = "landing";
export const OFFER_STANDARD_BUYER = "standardBuyer";
export const OFFER_STANDARD_SELLER = "standardSeller";
export const OFFER_AUTOBID_BUYER = "autobidBuyer";
export const OFFER_AUTOBID_SELLER = "autobidSeller";
export const BUY_TO_CONTRACT = "buyToContract";
export const SELL_TO_CONTRACT = "sellToContract";

// offer button status
export const EDIT_PREMIUM = "editPremium";
export const EDIT_SUBSCRIBE = "editSubscribe";
export const CREATE = "create";
export const ACCEPTED_IN_PROGRESS = "acceptedInProgress";
export const ACCEPTED_IN_CONTRACT = "acceptedInContract";
export const OPEN_IN_REVIEW = "openInReview";
export const DECLINED_ACCEPTED_IN_CONTRACT = "declinedAcceptedInContract";
export const DECLINED_BY_CLIENT = "declinedByClient";
export const DECLINED_JOB_DELETED = "declinedJobDeleted";
export const DECLINED_JOB_EXPIRED = "declinedJobExpired";
export const WITHDRAWN_BY_AGENT = "withdrawnByAgent";
