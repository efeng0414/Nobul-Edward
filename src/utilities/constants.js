import {
  OFFER_OPEN,
  OFFER_ACCEPTED,
  OFFER_NOT_ACCEPTED,
  OFFER_REJECTED,
  OFFER_JOBDELETED,
  OFFER_JOBEXPIRED,
  OFFER_WITHDRAWN
} from "../../core/constants/offers";

import { SELL, BUY } from "../../core/api-transform/jobs";
import { JOB_CLOSED } from "../../core/constants/jobs";

import {
  RESIDENTIAL as RESIDENTIAL_PROPERTY,
  COMMERCIAL as COMMERCIAL_PROPERTY,
  HOUSE_TOWNHOUSE as HOUSE_TOWNHOUSE_PROPERTY,
  CONDO_APARTMENT as CONDO_APARTMENT_PROPERTY,
  RECREATIONAL as RECREATIONAL_PROPERTY,
  OTHER as OTHER_PROPERTY
} from "../../core/constants/listings";

import {
  COMMERCIAL_BUY_ICON,
  COMMERCIAL_SELL_ICON,
  COMMERCIAL_GREY_ICON,
  CONDO_BUY_ICON,
  CONDO_SELL_ICON,
  CONDO_GREY_ICON,
  HOUSE_SELL_ICON,
  HOUSE_BUY_ICON,
  HOUSE_GREY_ICON,
  OTHER_BUY_ICON,
  OTHER_SELL_ICON,
  OTHER_GREY_ICON,
  RECREATIONAL_BUY_ICON,
  RECREATIONAL_SELL_ICON,
  RECREATIONAL_GREY_ICON,
  AVATAR_DEFAULT_LARGE,
  AVATAR_MALE_LARGE,
  AVATAR_FEMALE_LARGE,
  AVATAR_DEFAULT_SMALL,
  AVATAR_DEFAULT_MEDIUM,
  AVATAR_MALE_SMALL,
  AVATAR_MALE_MEDIUM,
  AVATAR_FEMALE_SMALL,
  AVATAR_FEMALE_MEDIUM,
  AVATAR_DEFAULT_SELECTED,
  AVATAR_MALE_SELECTED,
  AVATAR_FEMALE_SELECTED
} from "./images";

import ConsumerRegistrationForm from "../components/consumer-registration-steps/registration-form";
import ConsumerRegistrationConfirm from "../components/consumer-registration-steps/confirm";
import {
  AVATAR_TYPE_DEFAULT,
  AVATAR_TYPE_MALE,
  AVATAR_TYPE_FEMALE,
  AVATAR_DISPLAY,
  AVATAR_EDITING,
  AVATAR_SELECTED,
  AVATAR_SMALL
} from "../../core/constants/users";

export const LIMITED_LIST_DEFAULT_ELEMS = 3;

export const RESIDENTIAL = "residential";
export const COMMERCIAL = "commercial";
export const OTHER = "other";

export const HOME = "home";
export const INVESTMENT = "investment";

export const GOLD = "#ffc300";
export const WATER_BLUE = "#1283c6";

export const OFFER_STATUS_CLASSNAME = {
  [OFFER_OPEN]: "offer-status-open",
  [OFFER_ACCEPTED]: "offer-status-accepted",
  [OFFER_NOT_ACCEPTED]: "offer-status-no-accepted",
  [OFFER_REJECTED]: "offer-status-rejected",
  [OFFER_JOBDELETED]: "offer-status-job-deleted",
  [OFFER_JOBEXPIRED]: "offer-status-job-expired",
  [OFFER_WITHDRAWN]: "offer-status-withdrawn"
};

export const RegistrationStepMap = {
  0: ConsumerRegistrationForm,
  1: ConsumerRegistrationConfirm
};

export const HOME_PAGE_BUY_SELL_BG_CLASS = "blue";
export const HOME_PAGE_BROWSE_BG_CLASS = "purple";
export const HOME_PAGE_LEARN_BG_CLASS = "yellow";

export const CONUMER_REGISTRATION_STEP3_TITLE = [
  "consumer.contract_to_brokerage_title",
  "consumer.verification_required",
  "consumer.unverified_user_title"
];

export const CONUMER_REGISTRATION_STEP3_LINE1 = [
  "consumer.contract_to_brokerage_line1",
  "consumer.thank_you",
  "consumer.unverified_user_line_1"
];

export const CONUMER_REGISTRATION_STEP3_LINE2 = [
  "consumer.contract_to_brokerage_line2",
  "consumer.verification_send",
  "consumer.unverified_user_line_2"
];

export const PROPERTY_MAP = {
  [SELL]: {
    [RESIDENTIAL_PROPERTY]: HOUSE_SELL_ICON,
    [COMMERCIAL_PROPERTY]: COMMERCIAL_SELL_ICON,
    [HOUSE_TOWNHOUSE_PROPERTY]: HOUSE_SELL_ICON,
    [CONDO_APARTMENT_PROPERTY]: CONDO_SELL_ICON,
    [RECREATIONAL_PROPERTY]: RECREATIONAL_SELL_ICON,
    [OTHER_PROPERTY]: OTHER_SELL_ICON
  },
  [BUY]: {
    [RESIDENTIAL_PROPERTY]: HOUSE_BUY_ICON,
    [COMMERCIAL_PROPERTY]: COMMERCIAL_BUY_ICON,
    [HOUSE_TOWNHOUSE_PROPERTY]: HOUSE_BUY_ICON,
    [CONDO_APARTMENT_PROPERTY]: CONDO_BUY_ICON,
    [RECREATIONAL_PROPERTY]: RECREATIONAL_BUY_ICON,
    [OTHER_PROPERTY]: OTHER_BUY_ICON
  },
  [JOB_CLOSED]: {
    [RESIDENTIAL_PROPERTY]: HOUSE_GREY_ICON,
    [COMMERCIAL_PROPERTY]: COMMERCIAL_GREY_ICON,
    [HOUSE_TOWNHOUSE_PROPERTY]: HOUSE_GREY_ICON,
    [CONDO_APARTMENT_PROPERTY]: CONDO_GREY_ICON,
    [RECREATIONAL_PROPERTY]: RECREATIONAL_GREY_ICON,
    [OTHER_PROPERTY]: OTHER_GREY_ICON
  }
};

export const AVATAR_MAP = {
  [AVATAR_DISPLAY]: {
    [AVATAR_TYPE_DEFAULT]: AVATAR_DEFAULT_LARGE,
    [AVATAR_TYPE_MALE]: AVATAR_MALE_LARGE,
    [AVATAR_TYPE_FEMALE]: AVATAR_FEMALE_LARGE
  },
  [AVATAR_EDITING]: {
    [AVATAR_TYPE_DEFAULT]: AVATAR_DEFAULT_MEDIUM,
    [AVATAR_TYPE_MALE]: AVATAR_MALE_MEDIUM,
    [AVATAR_TYPE_FEMALE]: AVATAR_FEMALE_MEDIUM
  },
  [AVATAR_SMALL]: {
    [AVATAR_TYPE_DEFAULT]: AVATAR_DEFAULT_SMALL,
    [AVATAR_TYPE_MALE]: AVATAR_MALE_SMALL,
    [AVATAR_TYPE_FEMALE]: AVATAR_FEMALE_SMALL
  },
  [AVATAR_SELECTED]: {
    [AVATAR_TYPE_DEFAULT]: AVATAR_DEFAULT_SELECTED,
    [AVATAR_TYPE_MALE]: AVATAR_MALE_SELECTED,
    [AVATAR_TYPE_FEMALE]: AVATAR_FEMALE_SELECTED
  }
};

export const SERVICES_CHECKBOXES = "services-checkboxes";
export const SERVICES_GROUPS = "services-groups";
export const SERVICES_ACCORDION = "services-accordion";

export const PRICE_RANGE_SELECTOR_CLASS_NAME = "select-price-range";

export const locationSearchFilterOutTypes = [
  "room",
  "establishment",
  "finance",
  "food",
  "general_contractor",
  "place_of_worship",
  "street_number",
  "street_address",
  "subPremise",
  "natural_feature",
  "health"
];

export const minPasswordLength = 8;
export const maxPasswordLength = 16;

export const SELECT_CARD = "selectCard";
export const ADD_CARD = "addCard";

export const LANDING_PAGE_QUERY_PARAM = "ref=landing";
