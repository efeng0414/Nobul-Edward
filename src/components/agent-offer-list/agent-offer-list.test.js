import React from "react";
import AgentOfferListWrapped from "./index";
import toJson from "enzyme-to-json";
import { OFFER_OPEN, OFFER_ACCEPTED } from "../../../core/constants/offers";

describe("AgentOfferList", () => {
  const { WrappedComponent: AgentOfferList } = AgentOfferListWrapped;

  const mockGetOffersJobDetails = jest.fn();
  const componentProps = {
    intl: global.getIntl(),
    getOffersJobDetails: mockGetOffersJobDetails,
    offerList: [
      {
        status: "open"
      },
      {
        status: "withdrawn"
      },
      {
        status: "accepted"
      },
      {
        status: "withdrawn"
      }
    ],
    jobType: "buy",
    isPremium: true,
    getOfferCount: () => {},
    preventLoad: true
  };

  describe("Component Render", () => {
    const componentJSX = <AgentOfferList {...componentProps} />;

    const functionsToReset = ["handleStatusChange", "renderAgentOfferItem"];
    const componentPrototypeResets = functionsToReset.reduce(
      global.createComponentPrototypeResets({ component: AgentOfferList }),
      {}
    );

    const componentMount = Enzyme.shallow(
      componentJSX,
      global.getIntlContext({ intl: componentProps.intl })
    );

    beforeEach(() => {
      functionsToReset.map(
        global.resetPropertyComponentProperty({
          component: AgentOfferList,
          componentMount,
          componentPrototypeResets
        })
      );
    });

    it("should match existing snapshot", () => {
      expect(toJson(componentMount)).toMatchSnapshot();
    });

    describe("ComponentDidMount", () => {
      it("should call props.getOfferJobDetails on Mount", () => {
        expect(mockGetOffersJobDetails).toHaveBeenCalledWith({
          jobType: componentProps.jobType,
          offers: componentProps.offerList
        });
      });
    });

    describe("filterByStatus", () => {
      beforeEach(() => {
        componentMount.setState({ status: "", list: ["abc"] });
      });

      it("should set state correctly with status is shoWAll", () => {
        componentMount
          .find("Select")
          .at(1)
          .simulate("change", "showAll");
        const expectedOfferList = componentMount.instance().props.offerList;

        expect(componentMount.state().list).toEqual(expectedOfferList);
        expect(componentMount.state().status).toEqual("showAll");
      });

      it("should set state correctly with status is accepted", () => {
        componentMount
          .find("Select")
          .at(1)
          .simulate("change", OFFER_ACCEPTED);
        const expectedOfferList = {
          list: [{ status: OFFER_ACCEPTED }],
          status: OFFER_ACCEPTED
        };

        expect(componentMount.state()).toMatchObject(expectedOfferList);
      });

      it("should call props.getOfferCount", () => {
        const getOfferCountSpy = jest.fn();
        componentMount.setProps({ getOfferCount: getOfferCountSpy });
        componentMount
          .find(".offer-status-select")
          .at(0)
          .simulate("change", OFFER_OPEN);

        expect(getOfferCountSpy).toHaveBeenCalledWith({
          jobType: componentProps.jobType,
          offerCount: 1
        });
      });
    });

    describe("renderAgentOfferItem", () => {
      it("should call renderAgentOfferItem on render", () => {
        const renderAgentOfferItemMock = jest.fn();
        componentMount.instance().renderAgentOfferItem = renderAgentOfferItemMock;
        componentMount.setProps({ a: 1 });
        componentMount.find(".agent-list");

        expect(renderAgentOfferItemMock).toHaveBeenCalled();
      });

      it("should set JobType in AgentOfferItem correctly", () => {
        componentMount.setState({
          status: OFFER_OPEN,
          list: [{ status: OFFER_OPEN }]
        });

        expect(
          componentMount.find("Connect(InjectIntl(AgentOfferItem))").props()
            .jobType
        ).toBe("buy");
      });
    });
  });
});
