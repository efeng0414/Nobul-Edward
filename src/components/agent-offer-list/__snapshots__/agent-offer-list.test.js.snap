// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AgentOfferList Component Render should match existing snapshot 1`] = `
<div
  className="offer-list"
>
  <div
    className="offer-list-filter"
  >
    <Select
      choiceTransitionName="zoom"
      defaultValue="date"
      onSelect={[Function]}
      prefixCls="ant-select"
      showSearch={false}
      transitionName="slide-up"
    >
      <Option
        value="date"
      >
        Sort by date
      </Option>
    </Select>
    <Select
      choiceTransitionName="zoom"
      className="offer-status-select"
      defaultValue="Offer Status"
      onChange={[Function]}
      prefixCls="ant-select"
      showSearch={false}
      transitionName="slide-up"
    >
      <Option
        key="showAll"
        value="showAll"
      >
        Show all
      </Option>
      <Option
        key="offerOpen"
        value="open"
      >
        Open
      </Option>
      <Option
        key="offerAccepted"
        value="accepted"
      >
        Accepted
      </Option>
      <Option
        key="offerRejected"
        value="rejected"
      >
        Declined
      </Option>
      <Option
        key="offerWithdrawn"
        value="withdrawn"
      >
        Withdrawn
      </Option>
      <Option
        key="offerPendingVerification"
        value="pendingVerification"
      >
        Pending Verification
      </Option>
    </Select>
  </div>
  <div
    className="agent-list"
  >
    <Connect(InjectIntl(AgentOfferItem))
      className="agent-list-item"
      isPremium={true}
      jobType="buy"
      offerDetail={
        Object {
          "status": "open",
        }
      }
      preventLoad={true}
    />
    <Connect(InjectIntl(AgentOfferItem))
      className="agent-list-item"
      isPremium={true}
      jobType="buy"
      offerDetail={
        Object {
          "status": "withdrawn",
        }
      }
      preventLoad={true}
    />
    <Connect(InjectIntl(AgentOfferItem))
      className="agent-list-item"
      isPremium={true}
      jobType="buy"
      offerDetail={
        Object {
          "status": "accepted",
        }
      }
      preventLoad={true}
    />
    <Connect(InjectIntl(AgentOfferItem))
      className="agent-list-item"
      isPremium={true}
      jobType="buy"
      offerDetail={
        Object {
          "status": "withdrawn",
        }
      }
      preventLoad={true}
    />
  </div>
</div>
`;
